app/models/tenant.rb

```diff
@@ -39,8 +39,13 @@ def to_s
   attr_accessible :incorporated_on
 
   def fiscal_period(year)
-    month = fiscal_year_ends_on.try(:month) || 12
-    day = fiscal_year_ends_on.try(:day) || 31
+    if fiscal_year_ends_on && fiscal_year_ends_on.year == year
```

Comment:
I am not sure about this... What is the intention of `Tenant#fiscal_year_ends_on`?